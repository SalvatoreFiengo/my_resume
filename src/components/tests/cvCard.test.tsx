import CvCard from "../main/cvCard/cvCard";
import { shallow, mount } from '../../enzyme';

describe('cvCard',()=>{
    it('renders cvCard component without crashing',()=>{
        const setCvPosition = jest.fn();
        const cvCardShallowComponent = shallow(<CvCard setCvPosition={setCvPosition}/>);
        /*using getElements as html method gives error with styled components tags */
        expect(cvCardShallowComponent.getElements()).toMatchSnapshot();
    });
    it('Greets accordingly to current hour of the day',()=>{
        const setCvPosition = jest.fn();
        const cvCardShallowComponent = mount(<CvCard setCvPosition={setCvPosition}/>);
        const greeting = ()=>{
            const now = new Date().getHours();
            if(now < 12){
                return 'Good Morning'
            }
            else if(now < 19){
                return 'Good Afternoon'
            }
            else{
                return 'Good Evening'
            };
        };
        const greetingNode = cvCardShallowComponent.find('.custom-bottom-border > h1 > p');
        expect(greetingNode.text()).toEqual(greeting());
    });
    it('has ref once mounted',()=>{
        const setCvPosition = jest.fn();

        const cvCardShallowComponent = mount(<CvCard setCvPosition={setCvPosition}/>);
        const card=cvCardShallowComponent.find('div.card');
    });
})