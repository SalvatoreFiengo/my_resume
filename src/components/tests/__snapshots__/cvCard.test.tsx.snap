// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cvCard renders cvCard component without crashing 1`] = `
Array [
  <Styled(styled.div)
    className="card mx-auto mb-5"
  >
    <styled.div
      className="card-body"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-lg-6 col-xl-4"
        >
          <img
            alt="Salvatore Fiengo"
            className="w-100 d-block ps-3"
            src="https://media-exp1.licdn.com/dms/image/C5603AQHUrpA5rZ0-pA/profile-displayphoto-shrink_200_200/0/1550091903426?e=1625097600&v=beta&t=d-W-EDMy0rn1y-i4OIwIue8-mjbF6MtrUqfxY_Yf2xk"
          />
        </div>
        <div
          className="col-12 col-lg-6 col-xl-8"
        >
          <div
            className="custom-bottom-border mb-2 mt-3"
          >
            <h1
              className="mt-2 mb-2"
            >
              <p>
                Good morning
              </p>
              <span>
                I'm
              </span>
               
              Salvatore Fiengo
            </h1>
            <p
              className="mb-2"
            >
               
              Aspiring Developer
            </p>
          </div>
          <div
            className=""
          >
            <ul
              className=""
            >
              <li>
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-lg-3"
                  >
                    <p>
                      Address
                    </p>
                  </div>
                  <div
                    className="col-12 col-lg-9"
                  >
                    <p
                      className="detail"
                    >
                       
                      49 Shandon Mill, Shandon Park, Dublin 7
                    </p>
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-lg-3"
                  >
                    <p>
                      Email
                    </p>
                  </div>
                  <div
                    className="col-12 col-lg-9"
                  >
                    <p
                      className="detail"
                    >
                       
                      Sal.fiengo@gmail.com
                    </p>
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-lg-3"
                  >
                    <p>
                      Phone
                    </p>
                  </div>
                  <div
                    className="col-12 col-lg-9"
                  >
                    <p
                      className="detail"
                    >
                       
                      +353(0)834320505
                    </p>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </styled.div>
    <styled.div
      className="bg-success text-center d-flex justify-content-center"
    >
      <a
        aria-label="Github link here"
        className="m-2"
        href="https://github.com/SalvatoreFiengo/my_resume.git"
        rel="noreferrer"
        target="_blank"
      >
        <i
          aria-hidden="true"
          className="fa fa-2x fa-github text-white"
        />
      </a>
      <a
        aria-label="Linkedin Profile link here"
        className="m-2"
        href="https://www.linkedin.com/in/salvatorefiengo"
        rel="noreferrer"
        target="_blank"
      >
        <i
          aria-hidden="true"
          className="fa fa-2x fa-linkedin-square text-white"
        />
      </a>
    </styled.div>
  </Styled(styled.div)>,
]
`;
